src/test/kotlin/com/gildedrose/GildedRoseTest.kt

```diff
@@ -1,19 +1,168 @@
 package com.gildedrose
 
-import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.*
 import org.junit.jupiter.api.Test
 
 internal class GildedRoseTest {
 
     @Test
-    fun foo() {
-        val items = listOf(Item("foo", 0, 0))
+    fun normalUnexpiredItemDegrades() {
+        val items = listOf(
+            Item("Apples", 10, 50),
+            Item("Passes", 3, 20),
+            Item("Oranges", 1, 0)
+        )
         val app = GildedRose(items)
         app.updateQuality()
-        assertEquals("fixme", app.items[0].name)
+        val expected = listOf(
+            Item("Apples", 9, 49),
+            Item("Passes", 2, 19),
+            Item("Oranges", 0, 0)
+        )
+        for (i in expected.indices) {
+            assertEquals(expected[i].toString(), app.items[i].toString())
+        }
+    }
+
+    @Test
+    fun normalExpiredItemDegrades() {
+        val items = listOf(
+            Item("Apples", 0, 50),
+            Item("Passes", -2, 1),
+            Item("Oranges", -100, 0)
+        )
+        val app = GildedRose(items)
+        app.updateQuality()
+        val expected = listOf(
+            Item("Apples", -1, 48),
+            Item("Passes", -3, 0),
+            Item("Oranges", -101, 0)
+        )
+        for (i in expected.indices) {
+            assertEquals(expected[i].toString(), app.items[i].toString())
+        }
+    }
+
+    @Test
+    fun doNotDegradeSulfuras() {
+        val items = listOf(
+            Item("Sulfuras, Hand of Ragnaros", 0, 80), // TODO: match whole name?
+            Item("Sulfuras, Hand of Ragnaros", 10, 80),
+        )
+        val app = GildedRose(items)
+        app.updateQuality()
+        val expected = listOf(
+            Item("Sulfuras, Hand of Ragnaros", 0, 80),
+            Item("Sulfuras, Hand of Ragnaros", 10, 80),
+        )
+        for (i in expected.indices) {
+            assertEquals(expected[i].toString(), app.items[i].toString())
+        }
+    }
+
+    @Test
+    fun agedBrieIncreasesInQuality() {
+        val items = listOf(
+            Item("Aged Brie", 0, 0),
+            Item("Aged Brie", 5, 25),
+            Item("Aged Brie", 10, 50) // TODO: want case sensitive?
```

Comment:
Similarly here, what if we changed the logic for the max quality? That would break this entire test, is that useful?